lambdas/catalog-updater/__tests__/catalog-updater-test.js

```diff
@@ -0,0 +1,222 @@
+        // custom domain name swagger files have their api id and stage name in their filename
+        // in the future, we should vastly prefer this over way of recognizing the file over the other forms
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/api-gateway-swagger.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/a1b2c3d4e5_customdomain.json'})
+
+        expect(internalRepr.id).toBeUndefined()
+        expect(internalRepr.apiStageKey).toBe('a1b2c3d4e5_customdomain')
+        expect(internalRepr.generic).toBeUndefined()
+        expect(internalRepr.body).toEqual(JSON.parse(fileBody.toString()))
+    })
+
+    test('should recognize exported swagger files after fetching them from S3', async () => {
+        // 'normal' exported swagger files have their api id and stage name in their host and basepath fields
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/api-gateway-swagger.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/normal-exported-swagger.json'})
+
+        expect(internalRepr.id).toBeUndefined()
+        expect(internalRepr.apiStageKey).toBe('a1b2c3d4e5_exported')
+        expect(internalRepr.generic).toBeUndefined()
+        expect(internalRepr.body).toEqual(JSON.parse(fileBody.toString()))
+    })
+
+    test('should recognize generic OAS3 files after fetching them from S3', async () => {
+        // generic swagger is OAS3 not managed by api gateway; we detect this based on it not having any
+        // x-amazon-apigateway-integration fields in the path
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/generic-oas3.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/swagger.json'})
+
+        expect(internalRepr.id).toBe(index.hash('catalog/swagger.json'))
+        expect(internalRepr.apiStageKey).toBeUndefined()
+        expect(internalRepr.generic).toBe(true)
+        expect(internalRepr.body).toEqual(JSON.parse(fileBody.toString()))
+    })
+
+    test('should recognize custom domain name OAS3 files after fetching them from S3', async () => {
+        // custom domain name OAS3 files have their api id and stage name in their filename
+        // in the future, we should vastly prefer this over way of recognizing the file over the other forms
+
+        let fileBody = fs.readFileSync(path.join(__dirname, '/api-gateway-oas3.json')),
+            internalRepr
+
+        index.s3.getObject = jest.fn().mockReturnValueOnce(promiser({ Body: fileBody }))
+
+        internalRepr = await index.getSwaggerFile({Key: 'catalog/a1b2c3d4e5_oascustomdomain.json'})
```

Comment:
Update to`stageName` or something else that indicates that this is in fact a stage and not the custom domain name. (Part 2)